crates/yakui-widgets/src/widgets/textbox.rs

```diff
@@ -76,12 +79,28 @@ impl Widget for TextBoxWidget {
     fn update(&mut self, props: Self::Props) -> Self::Response {
         self.props = props;
 
-        let text = self.updated_text.as_ref().unwrap_or(&self.props.text);
+        let mut text = self.updated_text.as_ref().unwrap_or(&self.props.text);
+        let use_placeholder = text.is_empty();
+        if use_placeholder {
+            text = &self.props.placeholder;
+        }
 
         let mut render = RenderTextBox::new(text.clone());
         render.style = self.props.style.clone();
         render.selected = self.selected;
-        render.cursor = self.cursor;
+        if !use_placeholder {
+            render.cursor = self.cursor;
+        }
+        if use_placeholder {
+            // Dim towards background
+            render.style.color = Color::from_linear(
+                self.props
+                    .style
+                    .color
+                    .to_linear()
+                    .lerp(self.props.fill.unwrap_or(Color::CLEAR).to_linear(), 0.75),
```

Comment:
What do you think about adding `Color::lerp` to do the linear-srgb-linear conversions and make this a little more clear?

Reply:
Sure! Done.